src/onig-string-context.cc

```diff
@@ -19,3 +19,7 @@ OnigStringContext::OnigStringContext(Handle<String> str)
 bool OnigStringContext::IsSame(Handle<String> other) const {
   return v8String == other;
 }
+
+OnigStringContext::~OnigStringContext() {
+    v8String.Reset();
```

Comment:
We can use `Nan::Global` instead of `Persistent`, and `Nan::Global` calls `Reset` in its destructor so we won't need to call it here anymore.